### ISSUE TITLE
[nvfuser] view size is not compatible with input tensor's size and stride

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#1058]
Instructions 
* Uncompress Repro file - [repro.tar.gz](https://github.com/pytorch/torchdynamo/files/9436625/repro.tar.gz)
* Run `python repro/repro.py`

The repro is minified using https://github.com/pytorch/torchdynamo/pull/1056
This only fail with `aot_nvfuser` backend.

**Comment by rdspring1** [comment]
nvfuser is returning a contiguous tensor while eager mode creates a strided tensor.
The following `transpose` creates a strided tensor, which causes the runtime error with the final `linear` layer.

Here is standalone repro:
```
import torch
import torchdynamo
from torchdynamo.testing import rand_strided
from torchdynamo.debug_utils import run_fwd_maybe_bwd

torch._C._jit_set_nvfuser_single_node_mode(True)

from torch.nn import *
class ReproModule(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.linear_layer = Linear(196, 256)
        self.alpha = torch.randn(1, 1, 3, device='cuda')

    def forward(self, x : torch.Tensor):
        flatten_1 = torch.flatten(x, 2);
        transpose_2 = torch.transpose(flatten_1, 1, 2);

        # returns expected strided tensor like transpose_2
        # add_3 = torch.add(transpose_2, self.alpha)
        # returns contiguous tensor like alpha
        add_3 = torch.add(self.alpha, transpose_2)

        transpose_4 = torch.transpose(add_3, 1, 2);
        return self.linear_layer(transpose_4)

args = [((128, 3, 14, 14), (5376, 42, 14, 1), torch.float32, 'cuda', False)]
args = [rand_strided(sh, st, dt, de).requires_grad_(rg) for (sh, st, dt, de, rg) in args]

mod = ReproModule().cuda()
opt_mod = torchdynamo.optimize("aot_nvfuser")(mod)
res = run_fwd_maybe_bwd(opt_mod, args)
```

cc @jjsjann123

**Comment by jjsjann123** [comment]
yeah... this is expected behavior. A note is that we do intend to improve memory format support after we wrap up our transpose scheduler.

Currently we don't promise any memory format on output tensors, because the lack of performance transpose scheduler, permutation on output could bomb our kernel perf.

We intended to improve it by following TS profiled output memory format. I don't currently have a timeline for that, but the PR for permutation scheduler is getting there. https://github.com/csarofeen/pytorch/pull/1927. We'll still need to plumb it through after that PR. Likely separately for TS integration and nvprim.

**Comment by jjsjann123** [comment]
Just to clarify.
> nvfuser is returning a contiguous tensor while eager mode creates a strided tensor.
^^^ this is expected behavior in general. We try a best effort to stick to what eager has, but it's not a guarantee.


```
        flatten_1 = torch.flatten(x, 2);
        transpose_2 = torch.transpose(flatten_1, 1, 2);

        add_3 = torch.add(self.alpha, transpose_2)

        transpose_4 = torch.transpose(add_3, 1, 2);
        return self.linear_layer(transpose_4)
```

hmmm... looking at this example, I'm not sure how things get messed up with nvfuser. We are not fusing transpose, and our pointwise memory format handling should be pretty close to what eager has... But we also have `self.alpha = torch.randn(1, 1, 3, device='cuda')` which is just contiguous and it should dominate the output memory format.

We can take a look at the TS graph passed to nvfuser and figure out what went wrong there.
